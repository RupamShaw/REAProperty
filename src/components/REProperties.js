import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class REProperties extends Component {
    render() {
        const results = this.props.results
        const saved = this.props.saved
        console.log("results",results )
        console.log("saved",saved )
        return(
        <div>Hello from REA</div>
        )
    }
}
REProperties.propTypes = {
    results: PropTypes.array,
    saved:PropTypes.array
  }