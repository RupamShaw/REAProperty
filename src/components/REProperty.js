import React from 'react'
import PropTypes from 'prop-types'
import '../styles/REProperty.css'

const REProperty = (props) => {
    var cardProperty = props.cardProperty 
    var addREProperty = props.addREProperty 
    
     //dynamic style for button 
     var Button = ''
     var selectedPropertyColor = ''
     var backgroundColor = ''
 
     if (addREProperty) {
         Button = 'Add Property'
         selectedPropertyColor = 'green'
         backgroundColor = 'lightgreen'
     }

     const ButtonAStyle = {
        color: selectedPropertyColor,
        border: 'solid 2px ' + selectedPropertyColor,
        backgroundColor: backgroundColor
    }

    //dynamic style for each card of REPropertyData image headerbackgroundcolor and price 
    const HeaderBackgroundColorStyle = {
        backgroundColor: cardProperty.agency.brandingColors.primary
    }
    const mainImgUrl = cardProperty.mainImage;
    
    const MainImgStyle = {
        border: 'solid',
        margin: '5px',
        padding: '40px 0',
        backgroundImage: 'url("' + mainImgUrl + '")',
        backgroundSize: 'cover',
        backgroundPosition: 'center center',
    }

     // button add 
     const addProperty = () => {
        var cardProp = {
            "price": cardProperty.price,
            "agency": {
                "brandingColors": {
                    "primary": cardProperty.agency.brandingColors.primary
                },
                "logo": cardProperty.agency.logo
            },
            "id": cardProperty.id,
            "mainImage": cardProperty.mainImage
        }
        addREProperty(cardProp) //callback of parent
    }

    const onClick = e => {
        if (addREProperty) {
            addProperty()
        } 
    }

    return (
        <div className="propertyCardStyles" >
            <div id='backgrndColor' className="card-backgroundColor-header" style={HeaderBackgroundColorStyle}>
                <img src={cardProperty.agency.logo} className="card-logo" alt="logo" />
            </div>
            <div id='backgrndImg' style={MainImgStyle} >
               <div className="button"><a onClick={onClick} style={ButtonAStyle} href="#/"> {Button} </a></div>
           </div>
            <div style={{ borderTopStyle: 'solid' }}>
                <span style={{ marginLeft: '5px', float: 'left' }}>
                    Price
                 </span>
                <span id="price">
                    {cardProperty.price}
                </span>
            </div>
        </div>
    )
}

REProperty.propTypes = {
    addREProperty: PropTypes.func,
    removeProperty: PropTypes.func,
    cardProperty: PropTypes.shape({
            "price": PropTypes.string.isRequired,
            "agency":  PropTypes.shape({
                "brandingColors": {
                    "primary": PropTypes.string.isRequired
                }.isRequired,
            "logo": PropTypes.string.isRequired}),
            "id": PropTypes.string.isRequired,
            "mainImage":PropTypes.string.isRequired
        
        }).isRequired,
      
};

export default REProperty
