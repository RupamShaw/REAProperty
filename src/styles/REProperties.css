.propertyStyles {
    border: none;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    margin: 40px;
    justify-content: center;
}

.propertyTypes {
    border: none;
    max-width: 0%;
    min-width: 200px
}

.propertyListStyles {
    border: dotted;
    display: flex;
    flex-direction: column;
    max-width: 30%;
    min-width: 170px;
    margin: 10px;
    flex-wrap: wrap;
    justify-content: center;
}