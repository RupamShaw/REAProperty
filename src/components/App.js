import React, { Component } from 'react';
import '../styles/App.css';
import REProperties from './REProperties'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
           <h1 className="App-title">Welcome to REA World  via React</h1>
        </header>
        <REProperties  />
      </div>
    );
  }
}

export default App;
