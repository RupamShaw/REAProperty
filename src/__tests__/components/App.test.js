import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from '../../components/App';
import { REProperties }  from '../../components/REProperties';

const wrapper = shallow(<App />);

it('App renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.exists()).toEqual(true)
});
//mount doesn't work cause store
// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

it('renders the App title  ', (done) => {
  expect(wrapper.find('h1').text()).toEqual("Welcome to REA World  via React");
  done()
})

// it("REProperties Tag  should be there ", (done) => {
//   // REProperty tag totalcount is data.results+data.saved"
//   expect(wrapper.find('Connect(REProperties)').length).toEqual(1)
// })