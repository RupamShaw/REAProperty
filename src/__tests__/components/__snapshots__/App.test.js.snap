// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Welcome to REA World  via React
          </h1>
</header>,
        <Connect(REProperties) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            Welcome to REA World  via React
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to REA World  via React",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Welcome to REA World  via React",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Welcome to REA World  via React
            </h1>
</header>,
          <Connect(REProperties) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Welcome to REA World  via React
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to REA World  via React",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Welcome to REA World  via React",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
